### PR TITLE
Modern gem build and travis configuration

### PR DESCRIPTION
**Pull request by tpitale**
* use bundler
* minitest instead of test/unit
* simplify version to a string
* travis ci config
* remove jeweler

### CODE REVIEW

**Comment by tpitale** [comment]
Hooray for Travis!

File: lib/net/ssh/gateway/version.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module Net
+  module Ssh
+    module Gateway
+      VERSION = "1.2.0"
```
Comment (mfazekas) [inline lib/net/ssh/gateway/version.rb]: ```Ssh``` should be ```SSH``` and VERSION is likely 2.0.0 as the removal of Net::SSH::Gateway::Version is a non semver change

File: net-ssh-gateway.gemspec [hunk 2]
```diff
@@ -1,58 +1,28 @@
-# Generated by jeweler
-# DO NOT EDIT THIS FILE DIRECTLY
-# Instead, edit Jeweler::Tasks in Rakefile, and run 'rake gemspec'
-# -*- encoding: utf-8 -*-
+# coding: utf-8
+lib = File.expand_path('../lib', __FILE__)
+$LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
+require 'net/ssh/gateway/version'
```
Comment (mfazekas) [inline net-ssh-gateway.gemspec]: any reason for not using require_relative instead?

Comment (tpitale) [inline net-ssh-gateway.gemspec]: It's what the `bundle gem` command creates. Didn't modify it.

File: lib/net/ssh/gateway/version.rb (hunk 1 above)
Comment (tpitale) [inline lib/net/ssh/gateway/version.rb]: Will do.

Comment (tpitale) [inline lib/net/ssh/gateway/version.rb]: Also, assumed that you would bump the version before releasing the gem (if it was grouped with other changes). But, I can change it to 2.0.0.

**Comment by tpitale** [comment]
@mfazekas How does this look?

**Pull request merged**